Assets/Scripts/Enemy/Core/EnemyController.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyController : MonoBehaviour
+{
+    // PUBLIC DECLARATIONS
+    // Scripts
+    [Header("Entity Scripts")]
+    public EnemyBehaviour Behaviour;
+    public EnemyMovement Movement;
+    public EnemyRays Rays;
+    public EnemyAnimator Animator;
+    public EnemyMelee Melee;
+    public EnemyMeleeCharge MeleeCharge;
+    public EnemyMeleeChargedAttack MeleeChargedAttack;
+    public EnemyRanged Ranged;
+    public EnemyRangedChargedAttack RangedChargedAttack;
+
+
+    // Components
+    [Header("Components")]
+    public NavMeshAgent NavAgent;
+
```

Comment:
Decorators should be able to be written like 
```cs
[Header("Components")] public NavMeshAgent NavAgent;
```
This would mean it would be one line instead of many and you could remove the line breaks.